### ISSUE TITLE
Scalar Mix and Requires grad to PretrainedTransformerEmbedder

### ISSUE THREAD

**Issue by apmoore1** [allenai/allennlp#3462]
To use the [PretrainedTransformerEmbedder](https://github.com/allenai/allennlp/blob/master/allennlp/modules/token_embedders/pretrained_transformer_embedder.py#L9-L26) in a feature based/scalar mix approach rather than fine tuning the top layer which is only what you can do currently I believe. 

Further more I think the current implementation also does not consider the `attention_mask` that is an optional input in the forward function to alot of the HugginFace models e.g. the [BERT model](https://huggingface.co/transformers/model_doc/bert.html#transformers.BertModel.forward).

Solution:
To add these two arguments to the constructor: 
1. `top_layer_only` argument to state whether or not it should use either the top layer output like it currently does or whether it should use a `Scalar Mix`.
2. `requires_grad` if True all weights will be tuned else all weights will be frozen, currently this is always True.

Lastly check the type of  HugginFace model that is being used and see if an `attention_mask` is required and if so create an attention mask.

Reason:
I wanted to see the performance difference between a fine-tuned HugginFace model which is currently possible and the performance of a frozen/feature based HugginFace model but with a Scalar Mix weighting like in the [Deep contextualized word representations paper](https://www.aclweb.org/anthology/N18-1202.pdf). 

Happy to do the work and create the pull request.

**Comment by DeNeutoy** [comment]
Hi!

To keep things simple, I think we won't accept this contribution as it's not widely used. It should be straightforward to implement yourself though, as you have described.

Your point about the masking seems like a bug though. I'll open another issue for that.
